### PR TITLE
[Clang-tidy header][25/N] Fix clang-tidy warnings and enable clang-tidy on c10/cuda/*.{cpp,h}

### PR DESCRIPTION
**Pull request by cyyever**
This PR enables clang-tidy to code in c10/cuda.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: .lintrunner.toml [hunk 1]
```diff
@@ -204,9 +204,7 @@ exclude_patterns = [
     # CUDA files are also excluded.
     '**/fb/**',
     '**/*pb.h',
-    'c10/**/cuda/*pp',
     'aten/**/cuda/*pp',
-    '**/cuda/*pp',
```
Comment (Skylion007) [inline .lintrunner.toml:209]: Have we tested that these get scanned btw clang-tidy now? @malfet I think we added a PR to support lintrunner on cuda, but I don't remember the status.

Comment (cyyever) [reply .lintrunner.toml:209]: I am sure CUDA sources except 'aten/**/cuda/*pp' are scanned after this PR.

**Pull request closed without merging**